src/__tests__/renderHook.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import {renderHook} from '../pure'
+
+test('gives comitted result', () => {
+  const {result} = renderHook(() => {
+    const [state, setState] = React.useState(1)
+
+    React.useEffect(() => {
+      setState(2)
+    }, [])
+
+    return [state, setState]
+  })
+
+  expect(result.current).toEqual([2, expect.any(Function)])
+})
+
+test('allows rerendering', () => {
+  const {result, rerender} = renderHook(
+    ({branch}) => {
+      const [left, setLeft] = React.useState('left')
+      const [right, setRight] = React.useState('right')
+
+      // eslint-disable-next-line jest/no-if
+      switch (branch) {
+        case 'left':
+          return [left, setLeft]
+        case 'right':
+          return [right, setRight]
+
+        default:
+          throw new Error(
+            'No Props passed. This is a bug in the implementation',
+          )
+      }
+    },
+    {initialProps: {branch: 'left'}},
```

Comment:
In the spirit of creating a simple API, should we omit the `initialProps` functionality based on @mpeyper's [comment here](https://github.com/testing-library/react-testing-library/pull/991#issuecomment-966325340):

> I have long thought of intialProps to be a bit of a mistake and we've only kept it around for backwards compatibility. Having the callback accept no props and have variables for them be updated separately to rerender calls is not a huge overhead for the users and is easier for many to understand what is going on (less invisible lines between args and values)

I agree with @mpeyper's logic that this pattern is much more confusing than the alternative and would also enable this API to be even simpler in addition to being easier to use and understand (less footguns).